src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -0,0 +1,159 @@
+                };
+                var responseMessage = client.SendAsync(requestMessage).GetAwaiter().GetResult();
+                var responseString = responseMessage.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                var result = JsonConvert.DeserializeObject<Hashtable>(responseString);
+                if (responseMessage.IsSuccessStatusCode)
+                {
+                    WriteObject(result);
+                }
+                else
+                {
+                    WriteError(new ErrorRecord(new Exception(responseString), Guid.NewGuid().ToString(),
+                        ErrorCategory.InvalidResult, null));
+                }
+            }
+            catch (HttpRequestException httpRequestException)
+            {
+                WriteError(new ErrorRecord(httpRequestException, ErrorCategory.ConnectionError.ToString(),
+                    ErrorCategory.InvalidResult, null));
+                throw;
+            }
+            catch (Exception exception)
+            {
+                WriteError(new ErrorRecord(exception, ErrorCategory.ConnectionError.ToString(), ErrorCategory.InvalidOperation, null));
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Maps from HttpVerb to System.Net.Http.HttpMethod
+        /// </summary>
+        /// <returns>System.Net.Http.HttpMethod</returns>
+        private System.Net.Http.HttpMethod GetHttpMethod()
+        {
+            switch (Method)
+            {
+                case HttpVerb.GET:
+                    return System.Net.Http.HttpMethod.Get;
+                case HttpVerb.POST:
+                    return System.Net.Http.HttpMethod.Post;
+                case HttpVerb.PATCH:
+                    // System.Net.Http.HttpMethod does not contain HttpMethod.Patch in netstandard2.0
+                    // https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpmethod?view=netstandard-2.0
+                    return new System.Net.Http.HttpMethod("PATCH");
+                case HttpVerb.PUT:
+                    return System.Net.Http.HttpMethod.Put;
+                case HttpVerb.DELETE:
+                    return System.Net.Http.HttpMethod.Delete;
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        protected override void EndProcessing()
+        {
+            base.EndProcessing();
+        }
+
+        protected override void StopProcessing()
+        {
+            base.StopProcessing();
+        }
+
+        private void ValidateParameters()
+        {
+            if (Uri == null)
+            {
+                ThrowTerminatingError(new ErrorRecord(new ArgumentNullException(nameof(Uri), $"Must specify {Uri}"), Guid.NewGuid().ToString(),
+                    ErrorCategory.InvalidArgument, null));
+            }
+
+            if (Uri.IsAbsoluteUri && _client.BaseAddress.Host != Uri.Host)
```

Comment:
Make sure this `_client` is the same one used to make the request. See comment above.